core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -586,6 +592,19 @@ assert node(failedShard.currentNodeId()).getByShardId(failedShard.shardId()) ==
             " was matched but wasn't removed";
     }
 
+    private void promoteReplicaToPrimary(ShardRouting activeReplica, IndexMetaData indexMetaData,
+                                         RoutingChangesObserver routingChangesObserver) {
+        // if the activeReplica was relocating before this call to failShard, its relocation was cancelled above when we
```

Comment:
nit: this comment refers to "above" we should probably reword it or place it somewhere else.